network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -339,6 +391,12 @@ func (c *ControlMsgValidationInspector) processInspectMsgReq(req *InspectMsgRequ
 		c.metrics.AsyncProcessingFinished(req.validationConfig.ControlMsg.String(), time.Since(start))
 	}()
 
+	// iWant validation uses new sample size validation. This will be updated for all other control message types.
+	switch req.validationConfig.ControlMsg {
+	case p2pmsg.CtrlMsgIWant:
+		return c.inspectIWant(req.ctrlMsg.GetIwant())
```

Comment:
Please bear in mind that since `processInspectMsgReq` is invoked by the worker pool, any error it returns could lead to an irrecoverable failure, causing the node to crash—an outcome we certainly want to avoid. It appears the code is in this state due to it being a draft PR and will likely be updated in subsequent revisions. However, I thought it was important to underscore that `inspectIWant` returns at least two types of errors: `IWantCacheMissThresholdErr` and `DuplicateFoundErr`. For both of these errors, we should generate an `InvalidControlMessageNotification` and log the error, rather than returning it. Additionally, for all other types of errors returned, I propose we adopt a logging strategy, avoiding return unless a node crash is truly necessary.

Reply:
https://github.com/onflow/flow-go/pull/4574/commits/85f63c68646562b52de0832a59091c32fe196968